src/GraphQL/DataObjectType/PimcoreObjectType.php

```diff
@@ -92,11 +92,35 @@ public function build($context = [])
                 'args' => [
                     'keys' => [
                         'type' => Type::listOf(Type::string()),
-                        'description' => 'comma seperated list of key names'
+                        'description' => 'comma separated list of key names'
                     ]
                 ],
                 'resolve' => [$resolver, "resolveProperties"]
-            ]
+            ],
+            'parent' => [
+                'type' => $this,
```

Comment:
Isn't this way to restrictive ? Parent, sibling & children can be of any type (at least the ones enabled in the schema), right ?

Reply:
Funny thing is that this is possible (see demo.pimcore.fun) although the parent is a folder.
It returns the fullpath (I added it to the Car schema) but bombs out if I try to add a Car attribute :-)

{
  getCar(id: 260) {
  	name
    parent {      
      fullpath
    }    
  }
}

As said, parent should be a union type (allowed classes + folder) and children/siblings should be a list of that type.

Reply:
thanks for the feedback. I think you're right with using a union type (this restriction goes back when we designed the feature only with objects in mind).

Reply:
@AlsonPL almost resolved the union types issue (only an union type for documents/folders etc. is missing). We'll deliver this asap

Reply:
We're now able to get the parent, children and siblings on asset, object and documents, and also on asset folders, object folders and document folders.

Reply:
meaning that it would be ready to me merged assumed that everything is fine ? 😃

Reply:
I just added sample query to documentation. I hope everything is ok and it's ready to be merged :)